tests/components/elevenlabs/test_setup.py

```diff
@@ -0,0 +1,60 @@
+"""Tests for the ElevenLabs TTS entity."""
+
+from __future__ import annotations
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from httpx import ConnectError
+import pytest
+
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from .conftest import _client_mock
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_setup_async_client() -> Generator[AsyncMock]:
+    """Override async ElevenLabs client."""
+    with patch(
+        "homeassistant.components.elevenlabs.AsyncElevenLabs",
+        return_value=_client_mock(),
+    ) as mock_async_client:
+        yield mock_async_client
+
+
+@pytest.fixture
+def mock_setup_async_client_connect_error() -> Generator[AsyncMock]:
+    """Override async ElevenLabs client."""
+    client_mock = _client_mock()
+    client_mock.models.get_all.side_effect = ConnectError("Unknown")
+    with patch(
+        "homeassistant.components.elevenlabs.AsyncElevenLabs", return_value=client_mock
+    ) as mock_async_client:
+        yield mock_async_client
```

Comment:
So I think there's some value in reusing the same mock for both the config flow and normal usage. An example of this can be found in airgradient

Reply:
Thanks for pointing this out! I was not aware that this is possible - hopefully a neat solution now